### PR TITLE
create graph viewer

### PR DESCRIPTION
**Pull request by crockeo**
Looking into creating an interactive graph through [vello](https://github.com/linebender/vello)

current status:

image

### CODE REVIEW

File: src/shapes.rs [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+    [
+        Line::new(from, to),
+        Line::new(to, clockwise_arm),
+        Line::new(to, anticlockwise_arm),
+    ]
+}
+
+// TODO: is there something inside of one of the vello libraries
+// that would let us do this without having to implement it ourselves?
+struct Mat2([f64; 4]);
```
Comment (waywardmonkeys) [inline src/shapes.rs]: Would `kurbo::Affine` do what you need?

Comment (crockeo) [reply src/shapes.rs]: I think so, if I figure out the math :grin:. It didn't have `Mul` implemented with `Vec2` on the version I had originally used to write this--I'm not sure if it does now.

To me the simplest modeling of this was building a 2D vector which could be added to the point, so I just hacked together something that could do what I needed.

Comment (waywardmonkeys) [reply src/shapes.rs]: https://github.com/linebender/kurbo/issues/360

**Comment by crockeo** [comment]
Just merging this into the main branch, since it's working now! Needs some updates to build on others' machines (like pointing at my fork of Masonry in git, rather than on my computer :grin:), but it's good enough for now.

**Pull request merged**